src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,103 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.Objects;
```

Comment:
The use of `Objects` utility class is not allowed as per the checklist. Consider implementing equality checks manually.